core/src/replay_stage.rs

```diff
@@ -2824,6 +2849,30 @@ impl ReplayStage {
                     purge_repair_slot_counter,
                     SlotStateUpdate::BankFrozen(bank_frozen_state),
                 );
+                // If we previously marked this slot as duplicate in blockstore, let the state machine know
```

Comment:
Why would we not have returned replay results error abov, marked the slot dead, and let the state machine know there?

Reply:
This part is for blocks that we are able to freeze successfully. the handling on L2083 is for banks that are dead.